### PR TITLE
[colors] Add management of named Web colors to our CSS processing, prefix ANSI ones with `ansi_`

### LINKED ISSUE #442: [colours] Prefix ANSI colours with `ansi_`, and add management of named Web colours (which would then become the main source of named colours in CSS)

### PR DESCRIPTION
**Pull request by DrBenton**
Submitting this PR as a quick preview, since I realised that it brings an inconsistency issue between Rich and Textual: if we prefix all ANSI colour names with  `ansi_` in Textual we may have unexpected behaviours when we use the underlying Rich primitives in Textual - as Rich still have only ANSI colours, without the prefix in their names.

In the Textual DevTools for example, we use `padding = Segment("▌", Style.parse("bright_magenta"))` : I initially replaced _"bright_magenta"_ with _"ansi_bright_magenta"_ there, but doing so breaks the DevTools because the `Style` class is a Rich component, not a Textual one :grimacing: 

What do you folks think? @willmcgugan @darrenburns 

closes #442

### CODE REVIEW

**Comment by DrBenton** [comment]
@willmcgugan As discussed, I merged the first 16 ANSI colors with the named Web ones, and added the `transparent` pseudo-colour [eedec6c](https://github.com/Textualize/textual/pull/443/commits/eedec6c8630af01c83ae905ac6b45d8071faa782) :slightly_smiling_face:

File: src/textual/_color_constants.py [hunk 1]
```diff
@@ -1,239 +1,172 @@
 from __future__ import annotations
 
-ANSI_COLOR_TO_RGB: dict[str, tuple[int, int, int]] = {
+COLOR_NAME_TO_RGB: dict[str, tuple[int, int, int] | tuple[int, int, int, int]] = {
```
Comment (willmcgugan) [inline src/textual/_color_constants.py:3]: The alpha component is a float between 0 and 1

**Review by willmcgugan — CHANGES_REQUESTED** [review]
Great. Just a typing request.

File: src/textual/_color_constants.py (hunk 1 above)
Comment (willmcgugan) [reply src/textual/_color_constants.py]: NM fixed in merge

**Review by willmcgugan — APPROVED** [review]

**Comment by willmcgugan** [comment]
👍

**Pull request merged**